src/main/java/org/folio/circulation/services/AllowedServicePointsService.java

```diff
@@ -32,7 +60,92 @@ private CompletableFuture<Result<Map<RequestType, Set<String>>>> getAllowedServi
 
     log.debug("getAllowedServicePointsForItem:: parameters: request={}", request);
 
-    return ofAsync(new EnumMap<>(RequestType.class));
+    return userRepository.getUser(request.getRequesterId())
+      .thenCompose(r -> r.after(user -> lookupRequestPolicy(request, user)))
+      .thenCompose(r -> r.after(this::extractAllowedServicePoints));
+  }
+
+  private CompletableFuture<Result<RequestPolicy>> lookupRequestPolicy(
+    AllowedServicePointsRequest request, User user) {
+
+    log.debug("lookupRequestPolicy:: parameters: request={}, user={}", request, user);
+
+    if (user == null) {
+      log.error("lookupRequestPolicy:: user is null");
+      return completedFuture(failed(new RecordNotFoundFailure("User", request.getRequesterId())));
```

Comment:
Wouldn't it result in 404? This new endpoint should not return 404.

Reply:
changed to 422